### PR TITLE
Cookie Banner for All Apps

### PR DESCRIPTION
**Pull request by ertugrul-aydin**

### CODE REVIEW

File: packages/ui/src/components/CookieBanner/CookieBanner.tsx [hunk 1]
```diff
@@ -32,32 +31,25 @@ export const CookieBanner = (props: CookieBannerProps) => {
...
+        {''}
         <Navigate
           _hover={{ transform: 'scale(2.1)', color: 'primary.400' }}
-          href="#"
+          href="/privacy"
+          pl="2"
         >
-          Cookie Policy
+          {t('cookie.policy-url')}
         </Navigate>
```
Comment (7alip) [inline packages/ui/src/components/CookieBanner/CookieBanner.tsx:53]: We can use `<Trans />` component here.

Current translation:
`By using our website, you agree to the use of cookies as described in our Cookie Policy. <a>Cookie Policy<a/>`

With Trans:
`By using our website, you agree to the use of cookies as described in our <a>Cookie Policy</a>`

```tsx
<Trans 
  i18nKey={'cookie.text'} 
  components={{ a: <Navigate href="/privacy" /> }}
/>
```

In this case you can remove `cooki.policy-url` key from translations.

**Review by 7alip — CHANGES_REQUESTED** [review]
Thanks

File: packages/ui/src/components/CookieBanner/CookieBanner.tsx [hunk 2]
```diff
@@ -32,32 +31,25 @@ export const CookieBanner = (props: CookieBannerProps) => {
       bg="gray.700"
       rounded={'md'}
       boxShadow="sm"
-      {...rest}
+      position="fixed"
+      bottom="0"
+      right="0"
+      left="0"
```
Comment (7alip) [inline packages/ui/src/components/CookieBanner/CookieBanner.tsx:37]: If it's rounded, I prefer to have some space around the banner

```suggestion
      bottom="2"
      right="2"
      left="2"
```

File: packages/ui/src/components/CookieBanner/CookieBanner.tsx [hunk 3]
```diff
@@ -32,32 +31,25 @@ export const CookieBanner = (props: CookieBannerProps) => {
...
-          position="absolute"
-          right="2"
-          top="2"
-          color={'lightgray'}
-          onClick={onClose}
-        />
-      )}
-
       {!isMobile && (
         <Square size="12" bg="lightgray" borderRadius="md">
```
Comment (7alip) [inline packages/ui/src/components/CookieBanner/CookieBanner.tsx:40]: You can remove bg color

```suggestion
        <Square size="12" borderRadius="md">
```

**Review by 7alip — CHANGES_REQUESTED** [review]
Thanks

File: packages/ui/src/components/CookieBanner/CookieBanner.tsx [hunk 4]
```diff
@@ -32,32 +31,25 @@ export const CookieBanner = (props: CookieBannerProps) => {
       bg="gray.700"
```
Comment (7alip) [inline packages/ui/src/components/CookieBanner/CookieBanner.tsx:31]: Darker color would be better.

```suggestion
      bg="gray.900"
```

File: packages/ui/src/components/CookieBanner/CookieBanner.tsx (hunk 1 above)
Comment (ertugrul-aydin) [reply packages/ui/src/components/CookieBanner/CookieBanner.tsx:53]: Navigate is not working. In this way, text can not be clicked. 

`  <Text color="white" fontSize={{ base: 'sm', md: 'md' }}>
        <Trans
          i18nKey={'cookie.text'}
          components={{
            a: (
              <Navigate
                href={'/privacy'}
                _hover={{ transform: 'scale(2.1)', color: 'primary.400' }}
                pl="2"
              />
            ),
          }}
        />
      </Text>`

Comment (7alip) [reply packages/ui/src/components/CookieBanner/CookieBanner.tsx:53]: You might have forgotten to update the translations:

`Cookie Policy` should be wrapped with `a` tag: `"cookie.text": "... described in our <a>Cookie Policy</a>."`

Remember you should restart your dev server (`yarn foundation`) when you change your translations.

Please also remove `{t('cookie.policy-url')}`

Screenshot 2023-09-02 at 05 36 19

File: packages/ui/src/components/CookieBanner/CookieBanner.tsx [hunk 5]
```diff
@@ -68,30 +57,12 @@ export const CookieBanner = (props: CookieBannerProps) => {
           ...(isMobile && { width: '100%' }),
```
Comment (7alip) [inline packages/ui/src/components/CookieBanner/CookieBanner.tsx:57]: I overlooked this part.

Perhaps, the main reason of the hydration issue was using `isMobile`.

`isMobile` works only in the client side. We should avoid using client-only statements for styling or rendering parts. This usually trigger hydration error unless you use dynamic import.

`width` is core css attribute and could be used with regular responsive syntax:

`width={{ base: "full", sm: "auto" }}`

File: packages/ui/src/components/CookieBanner/CookieBanner.tsx [hunk 6]
```diff
@@ -29,36 +28,26 @@ export const CookieBanner = (props: CookieBannerProps) => {
...
-          right="2"
-          top="2"
-          color={'lightgray'}
-          onClick={onClose}
-        />
-      )}
-
       {!isMobile && (
-        <Square size="12" bg="lightgray" borderRadius="md">
+        <Square size="12" borderRadius="md">
```
Comment (7alip) [inline packages/ui/src/components/CookieBanner/CookieBanner.tsx:40]: The same `isMobile`.

We could use `display={{ ... }}`

File: packages/ui/src/components/CookieBanner/CookieBanner.tsx [hunk 7]
```diff
@@ -1,26 +1,25 @@
...
 
 export type CookieBannerProps = StackProps & {
-  onClose?: () => void
   onAllow?: () => void
-  onReject?: () => void
 }
 
-export const CookieBanner = (props: CookieBannerProps) => {
+const CookieBanner = (props: CookieBannerProps) => {
   const isMobile = useBreakpointValue({ base: true, md: false })
```
Comment (7alip) [inline packages/ui/src/components/CookieBanner/CookieBanner.tsx:20]: You can remove `isMobile`.

This breakpoint hook can be used in functions or some other situations which will not be triggered in the initial load.

**Review by 7alip — COMMENTED** [review]
Thanks. Just one thing about `isMobile`

**Review by 7alip — APPROVED** [review]
Thanks,  you can add the privacy pages in a separate branch later if you want. You can merge this

**Comment by 7alip** [comment]
I'll merge this to see if the same font loading issue will occur in dev deployment.

**Comment by 7alip** [comment]
@ertugrul-aydin Font issue seems to be fixed in dev. Feel free to create a new branch for privacy pages

**Pull request merged**